src/lib/containsEmail.js

```diff
@@ -0,0 +1,15 @@
+import assertString from './util/assertString';
+import isEmail from './isEmail';
+
+export default function containsEmail(str) {
+  assertString(str);
+  let n = str.length;
+  for (let i = 0; i < n; i++) {
+    for (let j = i + 1; j <= n; j++) {
+      if (isEmail(str.substring(i, j))) {
+        return true;
+      }
+    }
+  }
```

Comment:
I think this is super complex and not optimal. For sure it uses a vaidator.js function, but there are easier ways while looping through a text to get the emails.
For example we may want to check for the first @ character and then expand on both sides the search for a valid address.

Reply:
Yes, you are right I just didn't have any ideas at the moment and kinda just did it with the first thought. Gonna try and make it better soon.
Thanks for the comment and algorithm :)

Reply:
No problem mate!